pkg/controller/controller.go

```diff
@@ -830,19 +949,37 @@ func (c *controller) instanceDelete(obj interface{}) {
 // Binding handlers and control-loop
 
 func (c *controller) bindingAdd(obj interface{}) {
-	binding, ok := obj.(*v1alpha1.Binding)
-	if binding == nil || !ok {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+	if err != nil {
+		glog.Errorf("Couldn't get key for object %+v: %v", obj, err)
 		return
 	}
+	c.bindingQueue.Add(key)
+}
 
-	c.reconcileBinding(binding)
+func (c *controller) reconcileBindingKey(key string) error {
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+	binding, err := c.bindingLister.Bindings(namespace).Get(name)
+	if errors.IsNotFound(err) {
+		glog.Infof("Binding has been deleted %v", key)
```

Comment:
Suggest changing this message to `not doing work for binding %v because it has been deleted`